### PR TITLE
Scale expand param, add descriptions

### PR DESCRIPTION
**Pull request by OLarionova-HORIS**

### CODE REVIEW

File: plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt [hunk 1]
```diff
@@ -12,7 +12,23 @@ import jetbrains.datalore.plot.config.Option.Scale.DATE_TIME
...
+ * @param name string
+ *      The name of the scale - used as the axis label or the legend title. If None, the default, the name of the scale
+ *      is taken from the first mapping used for that aesthetic.
+ * @param breaks list of numeric
+ *     A numeric vector of positions (of ticks).
+ * @param labels list of strings
+ *      A vector of labels (on ticks).
+ * @param limits list of numerics
+ *      A numeric vector of length two providing limits of the scale.
+ * @param expand
```
Comment (alshan) [inline plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt:27]: List size == 1 ==> only multiplicative expand (and additive expand by default)

File: plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt [hunk 2]
```diff
@@ -12,7 +12,23 @@ import jetbrains.datalore.plot.config.Option.Scale.DATE_TIME
...
+ *      A numeric vector of length two giving multiplicative and additive expansion constants.
```
Comment (alshan) [inline plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt:28]: Document the defaults:
- continuous  scale: multiplicative = 0.05, additive = 0
- discrete scale: multiplicative= 0, additive = 0.6 
see: https://ggplot2.tidyverse.org/reference/scale_continuous.html

File: plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt [hunk 3]
```diff
@@ -34,3 +50,42 @@ fun scale_x_datetime(
...
+ * @param naValue
+ *       Missing values will be replaced with this value.
+ */
+fun scale_y_datetime(
+    name: String? = null,
+    breaks: List<Any>? = null,      // ToDo: should understand Date
+    labels: List<String>? = null,
+    limits: Pair<Number?, Number?>? = null,
+    expand: Any? = null,
```
Comment (alshan) [inline plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt:76]: type

File: plot-api/src/main/kotlin/jetbrains/letsPlot/scale/XYContinuous.kt [hunk 4]
```diff
@@ -31,7 +31,7 @@ fun scale_x_continuous(
     breaks: List<Number>? = null,
     labels: List<String>? = null,
     limits: Pair<Number?, Number?>? = null,
-    expand: Any? = null,
+    expand: List<Number>? = null,
```
Comment (alshan) [inline plot-api/src/main/kotlin/jetbrains/letsPlot/scale/XYContinuous.kt:34]: Let's change to `List<Any>` and add size and elements type checking similar to `scale_color_grey`

File: plot-api/src/main/kotlin/jetbrains/letsPlot/scale/XYContinuous.kt [hunk 5]
```diff
@@ -31,7 +31,7 @@ fun scale_x_continuous(
...
     naValue: Any? = null,
     trans: Any? = null
```
Comment (alshan) [inline plot-api/src/main/kotlin/jetbrains/letsPlot/scale/XYContinuous.kt:36]: Is this String? type? Let's fix in all scales.

File: plot-api/src/main/kotlin/jetbrains/letsPlot/scale/XYContinuous.kt (hunk 4 above)
Comment (OLarionova-HORIS) [reply plot-api/src/main/kotlin/jetbrains/letsPlot/scale/XYContinuous.kt]: Type was changed and checks were added

File: plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt (hunk 3 above)
Comment (OLarionova-HORIS) [reply plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt]: Done

File: plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt (hunk 2 above)
Comment (OLarionova-HORIS) [reply plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt:28]: Info about default values was added. Also need to add this link to the description?

File: plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt (hunk 1 above)
Comment (OLarionova-HORIS) [reply plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt]: Added to the description

File: demo/jvm-batik/src/main/kotlin/frontendContextDemo/scripts/LimitsContinuous.kt [hunk 6]
```diff
@@ -47,6 +45,22 @@ object LimitsContinuous {
...
+                    geom_vline(linetype = 3) { xintercept = -1 to 1 }
+                    ).show()
+
+            (p + scale_x_continuous(limits = -1 to 1, expand = listOf(.0, .5)) +
+                    geom_vline(linetype = 3) { xintercept = -1 to 1 }
+                    ).show()
+
+            (p + scale_x_continuous(limits = -1 to 1, expand = listOf(.0, .5)) +
+                    geom_vline(linetype = 3) { xintercept = -1 to 1 } +
+                    scale_y_continuous(limits = -2 to 2, expand = listOf(.0, 1.0)) +
```
Comment (alshan) [inline demo/jvm-batik/src/main/kotlin/frontendContextDemo/scripts/LimitsContinuous.kt:61]: Pls try if `expand = listOf(.0, 1)` works.

File: plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt (hunk 2 above)
Comment (alshan) [reply plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt:28]: No the link is mot needed.

File: plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt [hunk 7]
```diff
@@ -12,25 +12,94 @@ import jetbrains.datalore.plot.config.Option.Scale.DATE_TIME
...
-    breaks = breaks,
-    labels = labels,
-    limits = limits?.toList(),
-    expand = expand,
-    naValue = naValue,
-    otherOptions = Options(
-        mapOf(
-            DATE_TIME to true
+): Scale {
+    expand?.let { require(expand.size <= 2) { "'expand' can contain no more than two values: $expand" } }
```
Comment (alshan) [inline plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt:42]: How about `expand.size in (1.. 2)` ?

File: demo/jvm-batik/src/main/kotlin/frontendContextDemo/scripts/LimitsContinuous.kt (hunk 6 above)
Comment (OLarionova-HORIS) [reply demo/jvm-batik/src/main/kotlin/frontendContextDemo/scripts/LimitsContinuous.kt]: It works, example is updated.

File: plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt (hunk 7 above)
Comment (OLarionova-HORIS) [reply plot-api/src/main/kotlin/jetbrains/letsPlot/scale/DateTime.kt]: Updated

File: plot-api/src/main/kotlin/jetbrains/letsPlot/scale/ColorContinuous.kt [hunk 8]
```diff
@@ -33,7 +33,7 @@ fun scale_fill_continuous(
     expand: Any? = null,
```
[the remark this answers is not in the document]
Comment (OLarionova-HORIS) [reply plot-api/src/main/kotlin/jetbrains/letsPlot/scale/ColorContinuous.kt]: `expand` parameter is removed in non-positional scales.

**Pull request merged**